runtime/help/options.md

```diff
@@ -392,6 +392,11 @@ Here are the available options:
 
 	default value: `4`
 
+* `detectlimit`: if this is not set to 0, it will limit the automatic parsed
```

Comment:
IMHO I'd be rather reluctant to adding new options unless we really know there is a real need for it. (I.e. in this particular case, I'd rather stay with the hardcoded value of 100 for the time being.) Or the user will end up with a zillion options and get lost in them.

Reply:
To be honest I'm more a friend of having the choice rather been patronised. Just imagine a file with a very large comment block or other fancy header (larger than the e.g. hard coded 100) in front of the important decision mark. At the end the guess could be wrong. That was the reason for the option to give the choice to decide between more precision and faster parse time.

Reply:
Yeah, I can easily imagine a large comment block in the beginning of a file. Good point.

Having a choice is great, but I also think we should try our best to make software do the right thing from the beginning, rather than leave this burden to the user.

In this regard, now I'm thinking (independently of the question whether this line limit should be configurable or not): wouldn't it be a good idea to increase the default value from 100 to e.g. 500 right away? I actually doubt it would cause any real slowdown in practice (but if we concerned about that, we can always verify it with profiling, micro now has `-profile` command-line flag for that).

Reply:
BTW do you know how does it work in vim? i.e. how many lines are checked and how is it configured?

Reply:
Vim does it the same, but the limit(s) per different checks is/are hard coded (see https://github.com/vim/vim/blob/master/runtime/autoload/dist/ft.vim). The idea was good, but I thought going one step ahead is maybe better and we don't want to be Vim...it's `micro`. ;)